### PR TITLE
kvm: Security Group enhancements and refactor old code

### PR DESCRIPTION
**Pull request by wido**
## Description
This PR enhances the security group Python script. It refactors old code, removes deprecated code and fixes many bugs.

- PEP-8 styling
- Remove old LXC code from Python script
- Remove deprecated optparse from Python
- Use proper calls to iptables

## Types of changes
- [x] Cleanup (Code refactoring and cleanup, that may add test cases)

## How Has This Been Tested?
I have tested this on a Local Cloud we have running with the master branch.

Tested both IPv4 and IPv6 in a Advanced Zone using VXLAN L2 isolation.

### CODE REVIEW

**Comment by dhlaluku** [comment]
@blueorangutan package

File: scripts/vm/network/security_group.py [hunk 1]
```diff
@@ -35,11 +33,6 @@
 logpath = "/var/run/cloud/"        # FIXME: Logs should reside in /var/log/cloud
```
Comment (dhlaluku) [inline scripts/vm/network/security_group.py]: Since you are refactoring, should we perhaps fix the logpath as well?

**Comment by dhlaluku** [comment]
@blueorangutan test

File: scripts/vm/network/security_group.py (hunk 1 above)
Comment (wido) [reply scripts/vm/network/security_group.py]: I am not sure what the implications might be.

People might be parsing the files at that location, but also, if they are in /var/run/cloud they will be gone after a reboot (which is good!)

**Comment by rafaelweingartner** [comment]
@wido would you mind rebasing on top of current master?
Travis is presenting some odd results, which I think are not caused by your changes.

File: scripts/vm/network/security_group.py (hunk 1 above)
Comment (GabrielBrascher) [reply scripts/vm/network/security_group.py]: 5.0.0 sounds like a good plan.

**Comment by wido** [comment]
@rafaelweingartner I have just rebased against master and force pushed again

**Comment by GabrielBrascher** [comment]
@blueorangutan package

**Comment by dhlaluku** [comment]
@blueorangutan package

**Comment by dhlaluku** [comment]
@blueorangutan test

**Comment by GabrielBrascher** [comment]
Merging this one as the tests are looking good and we have 3 LGTMs.
Thanks for the PR @wido, and for the reviews @rafaelweingartner and @dhlaluku.

**Pull request merged**